common/remote/sockfile.cpp

```diff
@@ -1256,6 +1317,8 @@ protected: friend class CRemoteFileIO;
         : filename(_filename)
     {
         ep = _ep;
+        //Determine whether or not client should use SSL
+        querySecuritySettings(&useSSL, &configepPort, &clientCertificate.certificate, &clientCertificate.privateKey);
```

Comment:
This *may* be safe (because the values it sets will be the same), but this could be being called from several threads, simultaneously setting the internal members of clientCertificate ..

Would be better if you called and set in constructor clientCertificate, similar to the way you do in CSecuritySettings()